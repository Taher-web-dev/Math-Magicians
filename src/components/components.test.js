import React from 'react';
import renderer from 'react-test-renderer';
import Homepage from './Home';
import Quotepage from './Quote';
import CalculatorDesign from './Calculator';
import ResultIcone from './resultIcone';
import LigneOfButton from './ligneOfButton'

test('render correctly home page', () => {
  const tree = renderer.create(<Homepage />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly Quote page', () => {
  const tree = renderer.create(<Quotepage />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly ResultIcone component', () => {
  const title = 'Taher & Ugur'
  const tree = renderer.create(<ResultIcone />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('render correctly LigneOfButton component', ()=> {
  const props = {
      order : 2,
      handler : jest.fn()
  }
  it('render correctly when there is 3 buttons',()=> {
    props.textBtns = ['+', '-', '*'];
    const tree = renderer.create(<LigneOfButton
        textBtns={props.textBtns}
        order = {props.order}
        handler = {props.handler}
        />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('render correctly when there is 4 buttons',()=> {
    props.textBtns = ['+', '-', '*', '/'];
    const tree = renderer.create(<LigneOfButton
        textBtns={props.textBtns}
        order = {props.order}
        handler = {props.handler}
        />).toJSON();
    expect(tree).toMatchSnapshot();
  });
})