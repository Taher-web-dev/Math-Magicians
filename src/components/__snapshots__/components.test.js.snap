// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly LigneOfButton component render correctly when there is 3 buttons 1`] = `
<div
  className="command-line"
  style={
    Object {
      "gridRow": "2 / span 1",
    }
  }
>
  <div
    className="merge-btn"
    onClick={[MockFunction]}
  >
    <span>
      +
    </span>
  </div>
  <div
    className="simple-btn"
    onClick={[MockFunction]}
  >
    <span>
      -
    </span>
  </div>
  <div
    className="op-btn"
    onClick={[MockFunction]}
  >
    <span>
      *
    </span>
  </div>
</div>
`;

exports[`render correctly LigneOfButton component render correctly when there is 4 buttons 1`] = `
<div
  className="command-line"
  style={
    Object {
      "gridRow": "2 / span 1",
    }
  }
>
  <div
    className="simple-btn"
    onClick={[MockFunction]}
  >
    <span>
      +
    </span>
  </div>
  <div
    className="simple-btn"
    onClick={[MockFunction]}
  >
    <span>
      -
    </span>
  </div>
  <div
    className="simple-btn"
    onClick={[MockFunction]}
  >
    <span>
      *
    </span>
  </div>
  <div
    className="op-btn"
    onClick={[MockFunction]}
  >
    <span>
      /
    </span>
  </div>
</div>
`;

exports[`render correctly Quote page 1`] = `
<p
  style={
    Object {
      "color": "#4ab",
      "fontFamily": "Montserrat,sans-serif",
      "fontSize": "30px",
      "fontWeight": "bold",
      "lineHeight": "1.6",
      "marginLeft": "10%",
      "marginTop": "15%",
      "textAlign": "center",
      "width": "80%",
    }
  }
>
  Mathematics is not about numbers, equations computations,or algorithms: it is about understanding. -William Paul Thurston
</p>
`;

exports[`render correctly ResultIcone component 1`] = `
<div
  className="calculator-result"
>
   
  <span>
     
  </span>
   
</div>
`;

exports[`render correctly home page 1`] = `
<div
  style={
    Object {
      "margin": "5%",
      "width": "90%",
    }
  }
>
  <h2
    style={
      Object {
        "color": "rgb(133, 134, 147)",
        "fontFamily": "\\"Segoe UI\\", Arial, sans-serif",
        "marginBottom": "5%",
      }
    }
  >
     Welcome to our page! 
  </h2>
  <p
    style={
      Object {
        "color": "#4ab",
        "fontFamily": "Montserrat,sans-serif",
        "fontSize": "20px",
        "fontWeight": "bold",
        "lineHeight": "1.4",
        "textAlign": "center",
      }
    }
  >
    Everybody needs a Calculator at some point -- Full Screen, Fast Loading and FREE!
    <br />
    <span>
       Check it out! Math Magicians! 
    </span>
    <br />
    "Math Magicians" always available when you need to perform basic calculations such as addition, subtraction, multiplication, and division...
  </p>
</div>
`;
